### ISSUE TITLE
seeing frames of others' media

### ISSUE THREAD

**Issue by helenvarley** [upstage-org/mobilise#527]
often during the presentation i could see the frames of media items that were being operated by other players. usually (maybe always?) when this happened it meant i could not see or access the frame of the media item i was trying to operate.

players should ONLY see the frame of the media item that they are currently operating. no other frames should be visible.

@vickis did you also experience this?

**Comment by vickis** [comment]
it seems the frames are able to be selected (that they become visible
onMouse over...
this is especially when there is lots of media onstage

the issue is unheld media especially large ones which take a lot of screen
real estate

i am not sure what the fix is because the depth tool is really useful for
finding your own media but takes a few more clicks

**Comment by helenvarley** [comment]
how about this:

- frames NOT visible on mouse-over, only on click or from depth tool
- if another player is operating an object, the frame is not visible or accessible to other players (i guess like holding an avatar - should it be also that other objects - props, text, drawings etc - can also be held?)
- the only person who can see a media item's frame is the player who is holding or has selected (clicked on) that media item.

**Comment by vickis** [comment]
yes great :)

yes drawings (as an object can be exclusively held, moved and faded etc as
with other objects on stage)

**Comment by nguyenhongphat0** [comment]
I just digging into this issue, and realise that it was because you're trying to "operate" a prop while you are "holding" an avatar. Like this: you are Vicki, and want to move the heart, then add it to your avatar (as a costume). So the resizing frame should change to the prop.

<img width="557" alt="Screenshot 2021-09-02 at 17 47 19" src="https://user-images.githubusercontent.com/29925961/131830722-72170589-e63f-413c-a8f7-1587e1a9a868.png">

I guess the problem is that when the resizing frame is focused on the prop, the "left opacity slider" and the "right light bulb, remove button" should not be shown, correct? 🤔

**Comment by nguyenhongphat0** [comment]
I fixed this issue as mentioned above 👌 Let me know if the solution works for you 🙏

**Comment by vickis** [comment]
appears to be working! :)
will try and test with another player on stage in the next couple of days

**Comment by helenvarley** [comment]
yes this is looking much better thanks phat! it's now very clear that you are just getting the frame and tools for one media object at a time, & the red carrot shows you if you're still also holding an avatar :)

**Comment by helenvarley** [comment]
today i could see the frame of a text object held by another player. i was on the stage as a logged in player, but not holding any media. i checked and audience cannot see the frame, but it should be that you only see the frame of the object that you are operating. 
i was logged in in firefox, & looking at audience view in chromium.
![Screenshot from 2021-09-27 11:08:22](https://user-images.githubusercontent.com/3437324/134892392-78dba5c1-7bdd-4895-8e2d-da0be3a3a3bd.png)
![Screenshot from 2021-09-27 11:08:08](https://user-images.githubusercontent.com/3437324/134892399-7b7fbd57-4c96-46c7-9f71-ea5ca3c8cd40.png)
![Uploading Screenshot from 2021-09-27 11:03:19.png…]()

**Comment by helenvarley** [comment]
![Screenshot from 2021-09-27 11:01:43](https://user-images.githubusercontent.com/3437324/134892427-95227d26-5ad0-4ac9-87ee-2b02077f3905.png)

**Comment by helenvarley** [comment]
![Screenshot from 2021-09-27 11:03:19](https://user-images.githubusercontent.com/3437324/134892534-6e21023c-7c30-4cdf-b9c2-f1dd0138e8b2.png)
![Screenshot from 2021-09-27 11:03:06](https://user-images.githubusercontent.com/3437324/134892537-5c5649ca-8943-407e-9d90-45ed2786cc65.png)

**Comment by nguyenhongphat0** [comment]
Text object behave like a prop, it cannot be held by a specific player, so all the players could grab it and move/resize/delete it 🤔

**Comment by helenvarley** [comment]
we are still seeing frames of objects that have been held or selected by other players. today this was happening for me in firefox and chromium, and also for others in chrome and safari, but not for everyone was seeing others' frames. so it is not browser specific.
it should be that a player only ever sees the green frame of the object that they are currently holding (avatar) or have selected (prop, drawing, text).

the good news is that as far as we can tell the audience are not seeing any green frames, which is the most important thing. but it is distracting as a player to see others' frames.
